e2e/libs/setting/setting.py

```diff
@@ -26,26 +26,46 @@ def __init__(self):
         self.retry_count, self.retry_interval = get_retry_count_and_interval()
         self.backupstore = BackupStore()
 
+    def _update_setting(self, key, value):
+        setting = get_longhorn_client().by_id_setting(key)
+        get_longhorn_client().update(setting, value=value)
+
     def update_setting(self, key, value, retry=True):
+        logging(f"Trying to update setting {key} to {value} ...")
         if retry:
+            logging(
+                f"Retrying {self.retry_count} times with "
+                f"{self.retry_interval} seconds interval."
+            )
             for i in range(self.retry_count):
                 try:
-                    logging(f"Trying to update setting {key} to {value} ... ({i})")
-                    setting = get_longhorn_client().by_id_setting(key)
-                    get_longhorn_client().update(setting, value=value)
-                    if self.get_setting(key) == value:
-                        logging(f"Updated setting {key} to {value}")
-                        return
+                    self._update_setting(key, value)
+                    return
                 except Exception as e:
-                    logging(e)
-                time.sleep(self.retry_interval)
+                    logging(
+                        f"Retrying to update setting {key} to {value} ... "
+                        f"({i + 1})"
+                    )
+                    # Failed if it is the last retry
+                    if i == self.retry_count - 1:
+                        logging(
+                            f"Failed to update setting {key} to {value} after "
+                            f"{self.retry_count} attempts: {e}"
+                        )
+                        raise
+                    else:
+                        logging(
+                            f"Retrying to update setting {key} to {value} ... "
+                            f"(attempt {i + 1})"
+                        )
+                        time.sleep(self.retry_interval)
         else:
-            logging(f"Trying to update setting {key} to {value} ...")
-            setting = get_longhorn_client().by_id_setting(key)
-            get_longhorn_client().update(setting, value=value)
-            return
-
-        assert False, f"Failed to update setting {key} to {value} ... {setting}"
+            try:
+                self._update_setting(key, value)
+                logging(f"Successfully updated setting {key} to {value}.")
+            except Exception as e:
+                logging(f"Failed to update setting {key} to {value}: {e}")
+                raise
```

Comment:
It seems the `try/except/raise` is unnecessary, as re-raising the caught exception is the same as not having a `try/except` block at all.